Lib/test/test_inspect/test_inspect.py

```diff
@@ -1816,49 +1816,109 @@ def test_typing_replacement(self):
 class TestIsMethodDescriptor(unittest.TestCase):
 
     def test_custom_descriptors(self):
+        # Custom method descriptors (non-data descriptors):
+
         class MethodDescriptor:
+            """with __get__ and no __set__/__delete__"""
```

Comment:
Hm. I find attaching them directly to the classes they describe as most direct and readable.

However, if you insist they should be listed in the `for-loop`'s source collection, I'll move them.

Do you?

Reply:
Well, it's generally easier when you read the code so I'll leave that decision to the one merging your PR (I don't have those rights). For now, you can keep them as is I think.

Reply:
So, should those classes or their short descriptions (currently being their docstrings) stay as-is, or be moved/reorganized somehow?

Reply:
I personally don't like the descriptions at the docstring level because you need to go back and forth with the class whenever you see the actual test. Alternatively, you can just rename the class to include whatever is being done, namely:

- `MethodDescriptor -> MethodDescriptorWith_Get_NoSet_NoDelete`.
- `MethodDescriptorWithGetNone -> MethodDescriptor_GetIsNone_NoSet_NoDelete`.

Or, as I said before, do something like:

```py

class MethodDescriptor:
	def __get__(self, *_): pass

for cls, desc in [
	(MethodDescriptor, "__get__ and no __set__/__delete__"),
    ...
]:
	with self.subTest(f'ismethoddescriptor: {desc}', cls=cls):
		...
``` 

I see that you also use some classes twice but the tests have a different check (one is on `cls()` and the other is on `cls`. Since you don't include the docstring in the second test, you don't need to repeat `"__get__ and no __set__/__delete__"` the second time as well.

I think it's clearer that way, because when you read the test, you likely forget about the docstrings after you've read them, whereas when you read the list of classes being tested, it's easy to see which description correponds to which class (in addition, with a meaningful name, it's easier to remember when you use them twice).

One thing that I would say: try to keep the diff small (so if you can reduce the number of changes, it's better) and try to be consistent with what has been done (the tests use explicit assertion messages, and not docstrings, so I think there was a reason to).

Reply:
I have not hard feelings about this matter... @serhiy-storchaka What do you think?